### PR TITLE
Submodule system using hooks

### PR DESCRIPTION
**Pull request by snapwich**
## Type of change
- [x] Bugfix

## Description of change
Only include src dir and few node_modules inside of prebid-core bundle.

## Other information
related to #3908

### CODE REVIEW

**Comment by snapwich** [comment]
This partially fixes the problem by preventing modules from being bundled in prebid-core.  However it doesn't fix the submodule problem, and I'm still not sure exactly how to solve that problem.  With this merged modules won't end up in prebid-core, but submodules will still each have their own copy of the parent module, which is not ideal.

**Comment by snapwich** [comment]
Update the prebid-core hooks module with a `module` and `submodule` helper method which permits an install function for the parent module and the ability to pass data to the installer from the child module. This should allow for more sharing of functionality between child and parent modules without requiring either as static imports.

Also added the `./modules/.submodules.json` to make the gulp bundler aware of these relationships.  When using `gulp build --modules=digiTrustIdSystem` for example it will check the `.submodules.json` and see that digiTrust is a child of the `userId` module, so it will automatically include the parent module as well. Since they're using async hooks that are queued behind the scenes, load order of the modules doesn't matter.

**Comment by idettman** [comment]
Changed to on hold, since the ID5 User ID submodule has been merged, so I'm going to update this PR including changes to the ID5 submodule
'

**Comment by snapwich** [comment]
it might be worthwhile including the required changes to freewheel and adpod in this pull-request as well so we aren't left with a partial fix in master for module issues: @jsnellbaker @jaiminpanchal27

**Comment by jaiminpanchal27** [comment]
@snapwich I am starting that work today. I will update you as soon as it is done. Can I push to this branch or we create one new branch and create our PR's against that branch ?

**Comment by snapwich** [comment]
@jaiminpanchal27 you should be able to push to this branch, but a PR against this branch works as well.  Whatever is easier for you guys!

**Comment by snapwich** [comment]
@idettman don't forget you need to update `.submodules.json` with the new sub-module you added.

**Comment by idettman** [comment]
Working on an update to convert to prebidServer style module structure to allow imported sub modules

**Comment by idettman** [comment]
removed myself as a reviewer, since I've contributed a significant amount of code to this branch.

**Comment by jaiminpanchal27** [comment]
I have pushed adpod related changes. @jsnellbaker Please have a look and test 
Added on hold label so we do not merge this without testing all prebid submodules.

File: modules/.submodules.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "userId": [
+    "digiTrustIdSystem",
+    "pubCommonIdSystem",
+    "unifiedIdSystem",
+    "id5IdSystem",
+    "unifiedIdSystem"
```
Comment (snapwich) [inline modules/.submodules.json]: I think someone handled a merge improperly here.  Duplicate and removed modules are showing up now...

Might want to check to make sure no other changes were accidentally reverted as well.

Comment (idettman) [reply modules/.submodules.json]: Yeah this is supposed to be:
```
{
  "userId": [
    "digiTrustIdSystem",
    "id5IdSystem"
],
  "adpod": [
    "freeWheelAdserverVideo"
  ]
}
```

**Pull request merged**